Fixes links and details (#1288)
temporalio/documentation: docs/typescript/how-to-heartbeat-an-activity-in-typescript.md

---
id: how-to-heartbeat-an-activity-in-typescript
title: How to Heartbeat an Activity in TypeScript
sidebar_label: Activity Heartbeat
description: Heartbeat Activities to track their progress and get details of the Activity Execution.
tags:
  - developer-guide
  - sdk
  - typescript
---

> Long running activities should Heartbeat their progress back to the Workflow for earlier detection of stalled activities (with [Heartbeat Timeout](/concepts/what-is-a-heartbeat-timeout)) and resuming stalled activities from checkpoints (with Heartbeat details).

Author: ```suggestion
Long running Activities should Heartbeat their progress back to the Workflow for earlier detection of stalled activities (with [Heartbeat Timeout](/concepts/what-is-a-heartbeat-timeout)) and resuming stalled activities from checkpoints (with Heartbeat details).
```

> To set Activity Heartbet, use `Context.current().heartbeat()` in your Activity implementation, and set the `heartbeatTimeout` in your Workflow.

Author: ```suggestion
To set Activity Heartbeat, use `Context.current().heartbeat()` in your Activity implementation, and set the `heartbeatTimeout` in your Workflow.
```

> Long running Activities should Heartbeat their progress back to the Workflow for earlier detection of stalled activities (with [Heartbeat Timeout](/concepts/what-is-a-heartbeat-timeout)) and resuming stalled activities from checkpoints (with Heartbeat details).
> 
> To set Activity Heartbeat, use `Context.current().heartbeat()` in your Activity implementation, and set the `heartbeatTimeout` in your Workflow.

Reviewer: ```suggestion
Long-running Activities should Heartbeat their progress back to the Workflow for earlier detection of stalled Activities (with [Heartbeat Timeout](/concepts/what-is-a-heartbeat-timeout)) and resuming stalled Activities from checkpoints (with Heartbeat details).

To set Activity Heartbeat, use `Context.current().heartbeat()` in your Activity implementation, and set `heartbeatTimeout` in your Workflow.
```

```ts
// activity implementation
export async function example(sleepIntervalMs = 1000): Promise<void> {
  for (let progress = 1; progress <= 1000; ++progress) {
    await Context.current().sleep(sleepIntervalMs);
    // record activity heartbeat
    Context.current().heartbeat();
  }
}

//...

// workflow code calling activity
const { example } = proxyActivities<typeof activities>({
  startToCloseTimeout: '1 hour',
  heartbeatTimeout: '10s',
});
```

> In the previous example, setting the Heartbeat informs the Temporal Server of the Activity's progress regularly.

Author: ```suggestion
In the previous example, setting the Heartbeat informs the Temporal Server of the Activity's progress at regular intervals.
```
